Mage.Sets/src/mage/cards/m/MasterChef.java

```diff
@@ -0,0 +1,99 @@
+package mage.cards.m;
+
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.effects.common.continuous.GainAbilityAllEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.EntersTheBattlefieldEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+public class MasterChef extends CardImpl{
+
+    public MasterChef(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{G}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.BACKGROUND);
+
+        // Commander creatures you own have “This creature enters the battlefield with an additional +1/+1 counter on it” 
+        // and “Other creatures you control enter the battlefield with an additional +1/+1 counter on them.”
+        this.addAbility(new SimpleStaticAbility(new GainAbilityAllEffect(
+            new EntersBattlefieldAbility(
+                new AddCountersSourceEffect(CounterType.P1P1.createInstance()).setText("with an additional +1/+1 counter on it")
```

Comment:
Can you double check the text generation on this, I don't think it will work our properly.